centos5/Dockerfile

```
@@ -1,11 +1,16 @@
+## STAGE 1
+##   Bootstrap openssl, curl, git, file, which, xz, bzip2, tar, and portable-ruby
+##   Source for curl is copied in
 FROM centos:5
-LABEL maintainer="Shaun Jackman <sjackman@gmail.com>"
+LABEL maintainer="Maxim Belkin <maxim.belkin@gmail.com>"
 LABEL name="Linuxbrew/centos5"
 
 # Fix yum. CentOS 5 is end-of-life as of 2017-03-31.
-RUN urlgrabber https://gist.github.com/sjackman/132b65c33b62a89c45671e9c605025bc/raw/CentOS-Base.repo /etc/yum.repos.d/CentOS-Base.repo \
-	&& urlgrabber https://gist.github.com/sjackman/132b65c33b62a89c45671e9c605025bc/raw/libselinux.repo /etc/yum.repos.d/libselinux.repo \
-	&& yum install -y curl gcc gcc44 gcc44-c++ make sudo which \
+RUN sed -i \
+    '/^mirrorlist=/d; s|^#\(baseurl=http://\).*/\([^/]*\)/\$basearch/$|\1vault.centos.org/5.11/\2/$basearch|' \
+    /etc/yum.repos.d/{CentOS-Base.repo,libselinux.repo}
+
+RUN yum install -y {make,wget,gcc,perl}.x86_64 \
```

```suggestion
RUN yum install -y curl gcc make perl \
```
The`.x86_64` suffix should not be necessary. Let's use `curl` rather than `wget`. Put arbitrarily-ordered lists in alphabetical order.

---

`.x86_64` is there to not install 32 bit stuff. Without it, `yum install curl` translates to `yum install curl.i386 curl.x86_64` so you end up installing 32-bit stuff that is not necessary.

Why do you recommend using `curl` here?

---

> `.x86_64` is there to not install 32 bit stuff. Without it, `yum install curl` translates to `yum install curl.i386 curl.x86_64` so you end up installing 32-bit stuff that is not necessary.

I don't believe that's the case.

```sh
$ dnf install make
Installing:
 make                 x86_64        1:4.2.1-6.fc28         fedora         497 k
```

> Why do you recommend using `curl` here?

Brew depends on and uses `curl`, so let's use that rather than `wget` to avoid installing both.

---

Also for consistency.

---

Unfortunately, that's the case:
```
yum install curl
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Reducing CentOS-5 - libselinux to included packages only
Finished
Setting up Install Process
Resolving Dependencies
--> Running transaction check
---> Package curl.i386 0:7.15.5-17.el5_9 set to be updated
--> Processing Dependency: libc.so.6(GLIBC_2.4) for package: curl
--> Processing Dependency: libgssapi_krb5.so.2 for package: curl
--> Processing Dependency: libdl.so.2(GLIBC_2.1) for package: curl
--> Processing Dependency: libc.so.6(GLIBC_2.1.3) for package: curl
--> Processing Dependency: libc.so.6(GLIBC_2.2) for package: curl
--> Processing Dependency: libidn.so.11 for package: curl
--> Processing Dependency: libz.so.1 for package: curl
--> Processing Dependency: libc.so.6 for package: curl
--> Processing Dependency: libdl.so.2 for package: curl
--> Processing Dependency: libgssapi_krb5.so.2(gssapi_krb5_2_MIT) for package: curl
--> Processing Dependency: libc.so.6(GLIBC_2.3) for package: curl
--> Processing Dependency: libc.so.6(GLIBC_2.3.4) for package: curl
--> Processing Dependency: libk5crypto.so.3 for package: curl
--> Processing Dependency: libc.so.6(GLIBC_2.1) for package: curl
--> Processing Dependency: libssl.so.6 for package: curl
--> Processing Dependency: libcom_err.so.2 for package: curl
--> Processing Dependency: libcrypto.so.6 for package: curl
--> Processing Dependency: libc.so.6(GLIBC_2.0) for package: curl
--> Processing Dependency: libdl.so.2(GLIBC_2.0) for package: curl
--> Processing Dependency: libkrb5.so.3 for package: curl
---> Package curl.x86_64 0:7.15.5-17.el5_9 set to be updated
--> Processing Dependency: libidn.so.11()(64bit) for package: curl
--> Running transaction check
---> Package e2fsprogs-libs.i386 0:1.39-37.el5 set to be updated
--> Processing Dependency: libdevmapper.so.1.02 for package: e2fsprogs-libs
---> Package glibc.i686 0:2.5-123.el5_11.3 set to be updated
---> Package krb5-libs.i386 0:1.6.1-80.el5_11 set to be updated
--> Processing Dependency: libkeyutils.so.1 for package: krb5-libs
--> Processing Dependency: libselinux.so.1 for package: krb5-libs
--> Processing Dependency: libkeyutils.so.1(KEYUTILS_0.3) for package: krb5-libs
---> Package libidn.i386 0:0.6.5-1.1 set to be updated
---> Package libidn.x86_64 0:0.6.5-1.1 set to be updated
---> Package openssl.i686 0:0.9.8e-40.el5_11 set to be updated
---> Package openssl.x86_64 0:0.9.8e-40.el5_11 set to be updated
---> Package zlib.i386 0:1.2.3-7.el5 set to be updated
--> Running transaction check
---> Package device-mapper.i386 0:1.02.67-2.el5_11.1 set to be updated
--> Processing Dependency: libsepol.so.1 for package: device-mapper
---> Package keyutils-libs.i386 0:1.2-1.el5 set to be updated
---> Package libselinux.i386 0:1.33.4-5.7.el5.centos set to be updated
--> Running transaction check
---> Package libsepol.i386 0:1.15.2-3.el5 set to be updated
--> Finished Dependency Resolution

Dependencies Resolved

=======================================================================================================================================================
 Package                              Arch                         Version                                      Repository                        Size
=======================================================================================================================================================
Installing:
 curl                                 i386                         7.15.5-17.el5_9                              base                             235 k
 curl                                 x86_64                       7.15.5-17.el5_9                              base                             232 k
Installing for dependencies:
 device-mapper                        i386                         1.02.67-2.el5_11.1                           updates                          804 k
 e2fsprogs-libs                       i386                         1.39-37.el5                                  base                             120 k
 glibc                                i686                         2.5-123.el5_11.3                             updates                          5.4 M
 keyutils-libs                        i386                         1.2-1.el5                                    base                              18 k
 krb5-libs                            i386                         1.6.1-80.el5_11                              updates                          670 k
 libidn                               i386                         0.6.5-1.1                                    base                             194 k
 libidn                               x86_64                       0.6.5-1.1                                    base                             195 k
 libselinux                           i386                         1.33.4-5.7.el5.centos                        libselinux                        77 k
 libsepol                             i386                         1.15.2-3.el5                                 base                             128 k
 openssl                              i686                         0.9.8e-40.el5_11                             updates                          1.7 M
 zlib                                 i386                         1.2.3-7.el5                                  base                              51 k
Updating for dependencies:
 openssl                              x86_64                       0.9.8e-40.el5_11                             updates                          1.7 M

Transaction Summary
=======================================================================================================================================================
Install      13 Package(s)
Upgrade       1 Package(s)

Total download size: 11 M
```

---

dnf is fedora, not centos

---

I'm not able to reproduce this behaviour in CentOS either. I tested `centos:6` and `centos:7`. Which Docker image are you using?
```
yum install make
Installing:
 make           x86_64           1:3.82-23.el7             base           420 k
```

---

I tested `centos:5` as well, and it installs only the x86-64 package.
```
Installing:
 make           x86_64           1:3.81-3.el5              base           470 k
```

---

> Which Docker image are you using?

CentOS 5.

> I tested `centos:5` as well, and it installs only the x86-64 package.

Try installing curl, not make

---

You're right! That's weird.

```
Installing:
 curl               i386       7.15.5-17.el5_9             base           235 k
 curl               x86_64     7.15.5-17.el5_9             base           232 k
```

Here's a FAQ: https://wiki.centos.org/FAQ/General?action=show#head-357346ff0bf7c14b0849c3bcce39677aaca528e9

I suggest…
```sh
echo 'exclude = *.i?86' >>/etc/yum.conf
```

after which
```console
$ yum install curl
…
Installing:
 curl            x86_64          7.15.5-17.el5_9            base          232 k
```